initrd/bin/gui-init

```diff
@@ -174,10 +174,22 @@ update_totp()
   TRACE "Under /bin/gui-init:update_totp"
   # update the TOTP code
   date=`date "+%Y-%m-%d %H:%M:%S %Z"`
+  tries=0
   if [ "$CONFIG_TPM" != "y" ]; then
     TOTP="NO TPM"
   else
     TOTP=`unseal-totp`
+    # On platforms using CONFIG_BOOT_EXTRA_TTYS multiple processes may try to
+    # access TPM at the same time, failing with EBUSY. The order of execution
+    # is unpredictable, so the error may appear on main console, secondary one,
+    # or neither of them if the calls are sufficiently staggered. Try up to
+    # three times (including previous one) with small delays in case of error,
+    # instead of immediately scaring users with "you've been pwned" message.
+    while [ $? -ne 0 -a $tries -lt 2 ]; do
```

Comment:
I'm OK with this as-is, and I know we have this all over in Heads right, but since I saw it tweaked... :wink: `[ ... ] && [ ... ]` is better than `[ ... -a ... ]`.  `-a` introduces ambiguity if any of the variable arguments happen to be `!` or start with `-`.  The 2- and 3-argument forms of `[` are never ambiguous even if the variables contain stuff like that.

With that said, in this case the variables should never contain that stuff, and we have it all over in Heads already, so this is just a general comment for the future :grin:

Reply:
Changed. Thanks for explanation, I'm learning a lot today :slightly_smiling_face: